### ISSUE TITLE
Fix potential naming conflicts when users implement their reference FEs

### ISSUE THREAD

**Issue by fverdugo** [gridap/Gridap.jl#502]
**We definitively want to fix before releasing 0.15.0**

Say a user wants to implement his/her own version of a Lagrangian reference FE (just an example).

If he/she calls it `:Lagrangian` and defines the method 
```julia
ReferenceFE(p::Polytope,::Val{:Lagrangian},args...;kwargs...) where T 
```
There will be a potential name conflict with the version defined in Gridap. In other words, this is type piracy.

The situation can be even worse. Two users, A and B, implement a reference FE named `:MyRefFE` in their respective modules. Another user, C, wants to use the reference FEs implemented by A and by B. Name conflicts again. This is worse since neither A nor B will be aware of the name conflict. Only C when importing the modules of A and B.

The rood of the problem is that we are using `Val{<:Symbol}` to do the dispatch. This is dangerous. The solution is to use Julia types.
```julia
abstract type ReferenceFEName end
``` 
and then we do dispatch with specializations of this one. For instance, in Gridap we define
```julia
struct Lagrangian <: ReferenceFEName end
```
and then build reffes with 
```julia
reffe = ReferenceFe(Lagrangian(),Float64,order) 
```
instead of 
```julia
reffe = ReferenceFe(:Lagrangian,Float64,order) 
``` 

Just a very minor change in the user API.

Then any user is free to implement 
```julia
module MyModule
import Gridap
struct Lagrangian <: Gridap.ReferenceFEName end
end
``` 
in his/her own module, and it will be possible to use one or another in this elegant way:

```julia
reffe = ReferenceFe(Gridap.Lagrangian(),Float64,order)
reffe = ReferenceFe(MyModule.Lagrangian(),Float64,order) 
```

**We definitively want to implement this (5 minutes) before releasing 0.15.0**

**Comment by santiagobadia** [comment]
@fverdugo have you checked whether we have this very same problem in other parts of the code?
I can fix it next week, together with the missing renaming, if I have some time before :palm_tree:

**Comment by fverdugo** [comment]
> have you checked whether we have this very same problem in other parts of the code?

Let me take a look.

**Comment by fverdugo** [comment]
@santiagobadia I can take care of this issue. I am working on this in branch 

https://github.com/gridap/Gridap.jl/tree/fix_reffe_factory  (**not finished yet**)

Can you take care of the renaming `get_cell_data` -> `get_data`  of issue #480 ?

**Comment by fverdugo** [comment]
Fixed in PR #503.

I have also added and exported the constant
```julia
const lagrangian = Lagrangian()
```

so that we can use
```julia
reffe = ReferenceFe(lagrangian,Float64,order)
```
instead of
```julia
reffe = ReferenceFe(Lagrangian(),Float64,order)
```
